Pages/LanguagePage.cs

```diff
@@ -0,0 +1,113 @@
+﻿using OpenQA.Selenium;
+using SpecFlowMars.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SpecFlowProjectMars2.Pages
+{
+    public class LanguagePage : CommonDriver
+    {
+        public LanguagePage(IWebDriver driver) : base(driver)
+        {
+        }
+        public LanguagePage() { }   
+
+        //Web Elements
+
+        public IWebElement LanguageTab => driver.FindElement(By.XPath("//a[contains(text(),'Languages')]"));
+        public IWebElement AddNewButton => driver.FindElement(By.XPath("//*[@class='ui teal button ']"));
+        public IWebElement LanguageTextbox => driver.FindElement(By.XPath("//*[@placeholder='Add Language']"));
+
+
+        public IWebElement chooseLevelDropdown => driver.FindElement(By.XPath("//*[@class='ui dropdown']"));
+        public IWebElement chooseLevelOption => driver.FindElement(By.XPath("//*[@value='\" + Level + \"']"));
+
+
+        public IWebElement AddButton => driver.FindElement(By.XPath("//input[@value='Add']"));
+        public IWebElement CancelButton => driver.FindElement(By.XPath("//input[@value='Cancel']"));
+
+        public IWebElement ToolTipMessage => driver.FindElement(By.XPath("//*[@class='ns-box-inner']"));
+        public IWebElement EditLanguageTextbox => driver.FindElement(By.XPath("//*[@placeholder='Add Language']"));
+
+
+        public IWebElement EditChooselevelDropdown => driver.FindElement(By.XPath("\"//*[@class='ui dropdown']"));
+        public IWebElement EditpencilIcon => driver.FindElement(By.XPath("//div[@id='account-profile-section']//form//table//tbody[2]/tr/td[3]/span[1]/i"));
+
+
+        public IWebElement UpdateButton => driver.FindElement(By.XPath("//input[@value='Update']"));
+        public IWebElement EditChooseLevelOption => driver.FindElement(By.XPath("//*[@value='\" + newLevel + \"']"));
+        public IWebElement LastDeleteIcon => driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[3]/span[2]"));
+
+        public void Click_LanguageTab()
+        {
+            waitUtils.waitToBeVisible(driver, "Xpath", "LanguageTab", 10);
+            LanguageTab.Click();
+        }
+
+        public void CreateLanguageRecord(string Language, string Level)
+        {
+            waitUtils.waitToBeVisible(driver, "Xpath", "AddNewButton", 10);
```

Comment:
so you need to think about clean up and state 

example before adding language you need to make sure there a no existing languages as this could result in false positive test pass
you will need to think about this for all the scenarios

the state is say that you have a test that is adding a language english, the second time you run this test you should delete any existing language so that the state is reset, this will guarantee that your test does not produce false negatives and passes always
you can assume that test execution order is not always guaranteed
imagine if you run the tests  in parallel mode the all tests will run at the same time
so you need to also make sure you do not wipe out other tests data
hence why when you remove data the best thing to do is delete only data you added for the test to be correct
e.g test 1 added english and french as language test 2 added hindi then when you reset data you already know what language you are adding for each test
there for on reset of the two test you should delete english and french for test 1 and hindi for test 2

So to sum up you should delete all data if there is any before the start of any test and then after each test you should delete the data that a particular test is adding!
This is ensure that upon staring the test pack you have reset state and upon finishing each test deletes their data that they added!

also say you are updating a language then you must added language then update it do not assume the language is there!



so think about first clean up before any test runs (you delete all data around this feature e.g  before any language test run you delete all languages). Then after each test delete only data that the test has added! (test1 added english and test2 to added french then test1 should only delete english and test2 should delete french only)